cinema/views.py

```diff
@@ -1,6 +1,9 @@
+from django.db.models import Count, F, ExpressionWrapper, IntegerField
 from rest_framework import viewsets
+from django.utils.dateparse import parse_date
+from django.http import request
```

Comment:
The import statement `from django.http import request` is incorrect. Consider using `from rest_framework.request import Request` if you intend to use DRF's request object, or simply access `self.request` within viewsets.